src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using global::Iot.Units;
+
+namespace Iot.Device.Ad7193
+{
+    /// <summary>
+    /// Represents the Analog Devices AD7193, the 4-channel, 4.8 kHz, ultralow noise, 24-bit sigma-delta ADC with PGA
+    /// </summary>
+    public class Ad7193 : IDisposable
+    {
+        #region Metadata for IDevice
+
+        /// <summary>
+        /// Name of the manufacturer of the device
+        /// </summary>
+        public const string Manufacturer = "Analog Devices";
+
+        /// <summary>
+        /// Name of the product
+        /// </summary>
+        public const string Product = "AD7193";
+
+        /// <summary>
+        /// Category of the device
+        /// </summary>
+        public const string ProductCategory = "ADC";
+
+        /// <summary>
+        /// Description of the device
+        /// </summary>
+        public const string ProductDescription = "4-Channel, 4.8 kHz, Ultralow Noise, 24-Bit Sigma-Delta ADC with PGA";
+
+        /// <summary>
+        /// The URI of the datasheet of the device
+        /// </summary>
+        public const string DataSheetURI = "https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf";
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
+
+        /// <summary>
+        /// The maximum frequency that can be used on the SPI bus
+        /// </summary>
+        public const int MaximumSpiFrequency = 10000000;
+        #endregion
+
+        #region metadata for IAdcDevice
+
+        /// <summary>
+        /// Number of ADCs on the device
+        /// </summary>
+        public const int ADCCount = 1;
+
+        /// <summary>
+        /// The maximum bitrate of the ADC
+        /// </summary>
+        public const int ADCBitrate = 24;
+
+        /// <summary>
+        /// The maximum sampling rate of the ADC
+        /// </summary>
+        public const int ADCSamplerate = 4800;
+
+        /// <summary>
+        /// The number of channels the ADC has
+        /// </summary>
+        public const int ADCInputChannelCount = 8;
+        #endregion
+
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        // protected ConcurrentQueue<AdcValue> adcValues = new ConcurrentQueue<AdcValue>();
+        protected BlockingCollection<AdcValue> adcValues = new BlockingCollection<AdcValue>();
```

Comment:
Why protected if there are no subclasses to this type? It is better to have it as private for now, and then if we ever extend this to have subclasses we can change the visibility of the member

Reply:
This is protected, because I read in one of your contribution documents that you prefer to have a base class with the basic functionality of the driver, and let others extend on that class for the more device specific functions by inheriting from the driver. 
In this case it's important for some advanced functions to keep the access to the ADC values, I think, if someone would, let's say, implement the channel-separation/ordering logic for these values, to have separate buffers for every channel while using the multi-channel continuous conversion.

Reply:
I understand that, but seems to be that there are no subclasses today of this base class. My suggestion was that once somebody does indeed add a subclass from this base, then we can change it back to be protected, but there is no need to have it protected today if nobody is using it.

Reply:
Alright, I changed the scope of it to private.